doc/nrf/libraries/shell/shell_nfc.rst

```diff
@@ -0,0 +1,132 @@
+
+Requirements
+************
+
+This library has a following requirements:
+
+  * The :kconfig:option:`CONFIG_NFC_THREAD_CALLBACK` must be enabled.
+    The NFC callback function must be called from a thread context.
+    If default configuration it is called from the system workqueue context.
+    This library uses the operating system primitives in the callback function which cannot be
+    used in an interrupt context.
+  * If library logs and the logger are enabled, then the logger should not be used in the synchronous mode.
+    Synchronous logs processing can introduce to big latency in NFC events processing which causing the NFC shell backend being unstable.
+    If you need logs from this library, configure logger to be in the deferred mode by setting the :kconfig:option:`CONFIG_LOG_MODE_DEFERRED` Kconfig option.
+
+Configuration
+*************
+
+Set the :kconfig:option:`CONFIG_SHELL_NFC` Kconfig option to enable the shell backend.
+
+The following configuration options are available for this module:
+
+   * :kconfig:option:`CONFIG_SHELL_NFC_INIT_PRIORITY` sets initialization priority for the NFC shell backend.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_TX_BUFFER` sets the NFC T4T APDU response buffer size.
+     It size might depends on your Reader/Writer maxiumum supported APDU size.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_TX_RING_BUFFER_SIZE` sets the shell Tx ring buffer size.
+     It is used to buffer data coming from shell interface for the NFC T4T transport, that can send data only on the data reception from the Reader/Writer.
+   * :kconfig:option:`CONFIG_SHELL_NFC_BACKEND_RX_RING_BUFFER_SIZE`- sets the shell Rx ring buffer size.
+     It handles shell incoming data from the NFC transport.
+
+See the Kconfig help for details.
+
+Usage
+*****
+
+You can set the :kconfig:option:`CONFIG_SHELL_NFC` Kconfig option to enable the NFC shell.
+The shell is initialized automatically on the system initialization.
+You can approach your Reader/Writer device that uses the raw ISO-DEP (ISO14443-4A) protocol to start the shell interface.
+Next, forward the received data to the terminal and send back data received from terminal through your Reader/Writer device.
+For example, it can be the keyboard data or the bulk data.
+
+.. note::
+   The NFC transport can be physically disconnected and any time. After disconnection all data sent by the shell will be discarded.
+   The incomplete commands sent by the Reader/Writer before transport lost are discarded also.
+   Every transport connection or re-connection starts a new shell session.
+
+Samples using the library
+*************************
+
+The following |NCS| samples use this library:
+
+* :ref:`nfc_shell`
+
+Limitations
+***********
+
+The NFC T4T ISO-DEP transport is relatively slow.
+The maximum bit rates is 106 kbits/s for NFC-A technology.
+Data is sent in blocks and Reader/Writer device needs to send data first or send dummy data to poll data from the Tag.
```

Comment:
```suggestion
Data is sent in blocks.
The polling device needs to send data first or to send dummy data to poll data from the tag.
```
What is **poll data**?
Why we need info that data is sent in blocks?
Are they **blocks** indeed, or do you mean, e.g., messages or records?

Reply:
I mean poll for data :) ISO-DEP sends data in blocks. This is described in its specification